### ISSUE TITLE
Get-ADDomainController thows an error when AD domain contains RIVERBED DC lookalike objects

### ISSUE THREAD

**Issue by zjorz** [CarlWebster/Active-Directory-V3#15]
When riverbeds have been implemented within an enviroment, and that implementation contains DC lookalike objects, the Get-ADDomainController throws an error it cannot find the object. This is to be expected as it is not a real DC. I did not want to pay much attention to it. I would just mention that as "known issue"

**Comment by zjorz** [comment]
In addition, in the script log you will see 2 lines for each RIVERBED machine that looks like a DC....

PS>TerminatingError(Get-ADDomainController): "Directory object not found"
PS>TerminatingError(Get-ADDomainController): "Directory object not found"
PS>TerminatingError(Get-ADDomainController): "Directory object not found"
PS>TerminatingError(Get-ADDomainController): "Directory object not found"
PS>TerminatingError(Get-ADDomainController): "Directory object not found"
PS>TerminatingError(Get-ADDomainController): "Directory object not found"


in my case my demo/test AD has 3 RIVERBED devices that look like a DC, hence the 6 lines

**Comment by essentialexch** [comment]
Yeah, originally it crashed. :-) We ran into this at a client both Web and I worked at that has 6 Riverbeds. We considered replacing the code with something like the below, but I was not able to find a way to consistently exclude the Riverbeds but include both RODCs and RWDCs.

```
$DomainName = 'zz'

$sourceEntry        = New-Object System.Directoryservices.DirectoryEntry( "LDAP://$DomainName" )
$source             = New-Object System.Directoryservices.DirectorySearcher( $sourceEntry )
$source.SearchScope = 'Subtree'
$source.PageSize    = 1000

$machineAccount = '805306369'
$serverTrust    = ( 0x0080000 ).ToString()  # SERVER_TRUST_ACCOUNT
$trustDelegate  = ( 0x0002000 ).ToString()  # TRUSTED_FOR_DELEGATION
$partialSecrets = ( 0x4000000 ).ToString()  # PARTIAL_SECRETS_ACCOUNT (RODC)
$RWDCbitMask    = $serverTrust -bor $trustDelegate ## read-write DC account

#$groupDC   = 516
#$groupRODC = 521

#$source.Filter = "(&(sAMAccountType=$machineAccount)(|(primaryGroupId=$groupDC)(primaryGroupId=$groupRODC)))"
$source.Filter = "(&(sAMAccountType=$machineAccount)(|(userAccountControl:1.2.840.113556.1.4.803:=$RWDCbitMask)(userAccountControl:1.2.840.113556.1.4.803:=$partialSecrets)))"

$source.FindAll()

```

**Comment by zjorz** [comment]
I will check in my environment and see if I can exclude this somehow
